x-pack/test/functional/services/data_stream.ts

```diff
@@ -246,12 +246,21 @@ export function DataStreamProvider({ getService, getPageObject }: FtrProviderCon
 
     await updateDataStreamTemplate(streamIndex, mappings, mode);
 
-    log.info(`Creating ${streamIndex} data stream index...`);
-    await es.indices.createDataStream({
-      name: streamIndex,
-    });
-    // Uncomment only when needed
-    // log.debug(`PUT _data_stream/${streamIndex}`);
+    try {
+      log.info(`Creating ${streamIndex} data stream index...`);
+      await es.indices.createDataStream({
+        name: streamIndex,
+      });
+      // Uncomment only when needed
+      // log.debug(`PUT _data_stream/${streamIndex}`);
+    } catch (err) {
+      // If this specific error is met, delete it and try again
+      if (err.message.match(/resource_already_exists_exception/)) {
+        log.info(`Found resource_already_exists_exception: delete and rety again...`);
+        await deleteDataStream(streamIndex);
+        await createDataStream(streamIndex, mappings, mode);
+      }
```

Comment:
Sorry, I missed to add one suggestion here: if the error is not `resource_already_exists_exception` I think we should still throw the error so that we know if the data stream has failed to create.